src/main/java/uk/gov/hmcts/darts/audio/helper/UnstructuredDataHelper.java

```diff
@@ -0,0 +1,146 @@
+import uk.gov.hmcts.darts.common.repository.ExternalLocationTypeRepository;
+import uk.gov.hmcts.darts.common.repository.ExternalObjectDirectoryRepository;
+import uk.gov.hmcts.darts.common.repository.ObjectRecordStatusRepository;
+import uk.gov.hmcts.darts.common.repository.UserAccountRepository;
+import uk.gov.hmcts.darts.datamanagement.api.DataManagementApi;
+
+import java.time.OffsetDateTime;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
+
+import static uk.gov.hmcts.darts.common.enums.ExternalLocationTypeEnum.UNSTRUCTURED;
+import static uk.gov.hmcts.darts.common.enums.ObjectRecordStatusEnum.STORED;
+import static uk.gov.hmcts.darts.datamanagement.DataManagementConstants.MetaDataNames.MEDIA_REQUEST_ID;
+
+@Component
+@RequiredArgsConstructor
+@Slf4j
+public class UnstructuredDataHelper {
+    private final ExternalObjectDirectoryRepository externalObjectDirectoryRepository;
+    private final DataManagementApi dataManagementApi;
+    private final ObjectRecordStatusRepository objectRecordStatusRepository;
+    private final ExternalLocationTypeRepository externalLocationTypeRepository;
+    private final UserAccountRepository userAccountRepository;
+
+    private static final List<CompletableFuture> jobsList = new ArrayList<>();
+
+    public static List<CompletableFuture> getJobsList() {
+        return jobsList;
+    }
+
+    public boolean createUnstructured(ExternalObjectDirectoryEntity eodEntityToDelete, ExternalObjectDirectoryEntity eodEntity, BinaryData binaryData) {
+        boolean returnVal = true;
+        UUID uuid = saveToUnstructuredDataStore(eodEntity, binaryData);
+        if (uuid == null) {
+            returnVal = false;
+        } else {
+            saveToDatabase(eodEntity, uuid);
+            removeFromDatabase(eodEntityToDelete);
+        }
+        return returnVal;
+    }
+
+    private UUID saveToUnstructuredDataStore(ExternalObjectDirectoryEntity eodEntity, BinaryData binaryData) {
+        UUID uuid = null;
+        try {
+            Map<String, String> metadata = new HashMap<>();
+            metadata.put(MEDIA_REQUEST_ID, String.valueOf(eodEntity.getMedia().getId()));
+            BlobClient blobClient = dataManagementApi.saveBlobDataToContainer(binaryData, DatastoreContainerType.UNSTRUCTURED, metadata);
+            uuid = UUID.fromString(blobClient.getBlobName());
+            log.debug(
+                "Completed upload to unstructured data store for mediaRequestId {}. Successfully uploaded with blobId: {}",
+                eodEntity.getMedia().getId(),
+                uuid
+            );
+        } catch (Exception e) {
+            log.error(
+                "Upload to unstructured data store failed for mediaRequestId {}.",
+                eodEntity.getMedia().getId()
+            );
+        }
+        return uuid;
+    }
+
+    @Transactional
```

Comment:
I think these 2 private methods will need to be called from a public method with `@Transactional` to handle the composite save operations. The `@Transactional` can be removed from the private methods as they are redundant.